### ISSUE TITLE
MSO provider v0.3.2 is ignoring domain config:

### ISSUE THREAD

**Issue by soumukhe** [CiscoDevNet/terraform-provider-mso#74]
The code below is using remote user soumukhe  username / password which is the correct remote user associated with radius domain dmzrad and I can use that for UI login or for API code push.  However this is not working and gives me a credentials error.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### MSO version

* V x.x.x

### APIC version and APIC Platform for Site Level Resources

* V x.x.x and on-prem/cloud-aws/cloud-azure/all.

### Affected Resource(s)

* mso_XXXXX

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by soumukhe** [comment]
# Versions:
ND: 2.1.1d
NDO : 3.5.1e
Terraform:  v1.0.10
mso_plugin: v0.3.2

# provider config:

provider "mso" {
  username = "soumukhe" # <MSO username>
  password = "soumuSekret1@!" # <MSO pwd>
  url      = "https://10.0.0.94/" # <MSO URL>
  domain   = "dmzrad"
  insecure = true
  platform = "nd"
}

# output of Terraform Apply:

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

mso_tenant.tenant02: Creating...
╷
│ Error: Invalid Username or Password
│ 
│   with mso_tenant.tenant02,
│   on main.soumukhe.tf line 18, in resource "mso_tenant" "tenant02":
│   18: resource "mso_tenant" "tenant02" {
│ 
# Comment:
When I change the username to admin with admin password, it works fine.  There is a admin local user and admin remote user with the same password.  Obviously, it's ignoring the domain key and just looking at the local user credentials.

**Comment by lhercot** [comment]
This issue should be fixed in v0.3.3. Thanks for reporting it.

**Comment by soumukhe** [comment]
This is still not working for remote user with 0.3.3.

```ccl
aciadmin@ubuntu-jump:~/Terraform/AWS-EKS-Terraform/AWS_Ter_EKSCTL/childNDO3.6$ terraform plan
╷
│ Error: Invalid Username or Password
│ 
│   with data.mso_site.aws_site10,
│   on resources.tf line 1, in data "mso_site" "aws_site10":
│    1: data "mso_site" "aws_site10" {
│ 
```
